core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -454,6 +454,9 @@ class AdminManager(val config: KafkaConfig,
   private def alterTopicConfigs(resource: ConfigResource, validateOnly: Boolean,
                                 configProps: Properties, configEntriesMap: Map[String, String]): (ConfigResource, ApiError) = {
     val topic = resource.name
+    if (!metadataCache.contains(topic))
+      throw new UnknownTopicOrPartitionException(s"The topic '$topic' does not exist.")
+
     adminZkClient.validateTopicConfig(topic, configProps)
```

Comment:
It seems not all checks in ```validateTopicConfig``` are required. 

1. ```Topic.validate(topic)``` -> the topic is validated already as the topic is created
1. ```if (!zkClient.topicExists(topic))``` -> it is replaced by ```if (!metadataCache.contains(topic))```
1. ```LogConfig.validate(configs)``` -> this should be reserved

In short, should we remove ```adminZkClient.validateTopicConfig(topic, configProps)``` and then add ```LogConfig.validate(configs)```?

Reply:
hmmm, the following ```changeTopicConfig``` checks the topic name and configs again.  Not sure why there are such duplicate checks.

Reply:
yeah, there is lot scope for cleanup. we can improve future PRs.